modules/capture-admin-service-impl/src/main/java/org/opencastproject/capture/admin/impl/AgentImpl.java

```diff
@@ -96,10 +98,15 @@
 
   /** The roles allowed to schedule this agent */
   @ElementCollection
-  @Column(name = "role")
-  @CollectionTable(name = "oc_capture_agent_role", joinColumns = { @JoinColumn(name = "id", referencedColumnName = "id"),
-          @JoinColumn(name = "organization", referencedColumnName = "organization") })
-  protected Set<String> schedulerRoles = new HashSet<String>();
+  @Column(name = "role", nullable = false)
+  @CollectionTable(name = "oc_capture_agent_role", indexes = {
+      @Index(name = "IX_oc_capture_agent_role_pk", columnList = "id, organization")
+    }, uniqueConstraints = {
+      @UniqueConstraint(name = "UNQ_capture_agent_role_id_org_role", columnNames = {"id", "organization", "role"})
+    }, joinColumns = {
+      @JoinColumn(name = "id", referencedColumnName = "id", nullable = false),
+      @JoinColumn(name = "organization", referencedColumnName = "organization", nullable = false) })
```

Comment:
_Comment to myself: Next time, respond to reviews earlier. That makes it easier to remember what you did._ Nevertheless, the new constraint makes sense since we are joining on the id and organization.